openfe/protocols/openmm_rfe/_rfe_utils/relative.py

```diff
@@ -229,6 +232,164 @@ def __init__(self,
         self._omm_hybrid_topology = self._create_hybrid_topology()
         logger.info("Hybrid system created")
 
+    @staticmethod
+    def _verify_cmap_compatibility(
+            cmap_old: openmm.CMAPTorsionForce | None,
+            cmap_new: openmm.CMAPTorsionForce | None,
+    ) -> tuple[
+        openmm.CMAPTorsionForce | None,
+        openmm.CMAPTorsionForce | None,
+        int,
+        int,
+        int,
+        int,
+    ]:
+        """
+        Verify CMAPTorsionForce compatibility between two systems.
+
+        Parameters
+        ----------
+        cmap_old : openmm.CMAPTorsionForce | None
+           CMAPTorsionForce from the old system, or ``None`` if absent.
+        cmap_new : openmm.CMAPTorsionForce | None
+           CMAPTorsionForce from the new system, or ``None`` if absent.
+
+        Returns
+        -------
+        tuple
+           (cmap_old, cmap_new, old_num_maps, new_num_maps, old_num_torsions, new_num_torsions)
+           where the first two entries are the input forces (possibly ``None``)
+           and the remaining four are integers describing the number of maps and
+           torsions in each force.
+
+        Raises
+        ------
+        RuntimeError
+           If only one of the forces is present, or if the number of maps or the
+           number of torsions differs between the two forces.
+        """
+        # if only one has cmap raise an error
+        if (cmap_new is None) ^ (cmap_old is None):
+            raise RuntimeError(f"Inconsistent CMAPTorsionForce between end states expected to be present in both"
+                               f"but found in old: {bool(cmap_old)} and new: {bool(cmap_new)}")
+
+        if cmap_new == cmap_old is None:
```

Comment:
You check this both here and [here](https://github.com/OpenFreeEnergy/openfe/pull/1695/files#r2566139589), so this code path will likely not be hit by tests.